server/mrt_converter/mrt_converter.go

```diff
@@ -0,0 +1,95 @@
+package converter
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/osrg/gobgp/pkg/packet/bgp"
+	"github.com/osrg/gobgp/pkg/packet/mrt"
+)
+
+type update struct {
+	Collector string
+	SeenAt    time.Time
+	PeerAS    uint32
+
+	// Data inside BGP updates.
+	Announced  []*bgp.IPAddrPrefix
+	Withdrawn  []*bgp.IPAddrPrefix
+	Attributes map[bgp.BGPAttrType]bgp.PathAttributeInterface
+}
+
+// extractCollector extracts the collector name from the input file path.
+func extractCollector(filename string) (string, error) {
+	if filename == "" {
+		return "", fmt.Errorf("empty file path")
+	}
+	if filename[0] != '/' {
+		return "", fmt.Errorf("path must starts with '/'")
+	}
+	if !strings.Contains(filename, "bgpdata") {
+		return "", fmt.Errorf("file %s is not a valid RouteViews archive path", filename)
+	}
+
+	dirs := strings.Split(filename, "/")
+	if dirs[1] == "bgpdata" {
```

Comment:
this seems to assume 'we are always getting `route-views2` as a collector"

Is that the case? Longer term I think we'd be getting data from all the RV collectors, and collectors from RIPE as well, where their path structure is different.

again, a:
   TODO():fix when we have more collectors

would be totally fine.

Reply:
Understood. If we want to (1) get data from other sources and (2) trigger conversions through GCS uploads, we will need to  be able to differentiate sources based on the paths, and at that time we will certainly need to have different collector name extraction functions for different sources.